logging/logrus/interceptors_test.go

```diff
@@ -214,3 +218,82 @@ func (s *LogrusLoggingSuite) TestPingList_WithCustomTags() {
 	assert.Contains(s.T(), msgs[1], `"level": "info"`, "OK error codes must be logged on info level.")
 	assert.Contains(s.T(), msgs[1], `"grpc.time_ms":`, "interceptor log statement should contain execution time")
 }
+
+func TestLogrusLoggingOverrideSuite(t *testing.T) {
```

Comment:
I've rebased these changes on top of your refactor, but have kept the new suite(s) for now pending clarification

With regards to your suggestion to use an override function, do you mean something like this, or have I misunderstood?:

```
opts := []grpc_logrus.Option{
	...
	grpc_zap.WithDurationField(func(duration time.Duration) zapcore.Field {
		if s.durationOverrideFunc != nil {
			return s.durationOverrideFunc(duration)
		}
		return grpc_zap.DefaultDurationToField(duration)
	}),
}
...
type zapServerSuite struct {
	*zapBaseSuite
	durationOverrideFunc grpc_zap.DurationToField
}
```

With the above, verifying behavior of the default options (i.e. that `grpc.time_ms` is used when not overridden) doesn't happen since the default option is always replaced, even if it ultimately resolves to the same function call